### PR TITLE
Add Power Shell script for lib copying on Windows

### PR DESCRIPTION
**Pull request by infomiho**
## Description

We added a new bash script that build the libs and copies them in the `data` dir. This PR adds the same script but in Power Shell for Windows support.

## Type of change

- [x] **🔧 Just code/docs improvement** 
- [ ] **🐞 Bug fix** 
- [ ] **🚀 New/improved feature** 
- [ ] **💥 Breaking change** 

## Checklist

- [ ] I tested my change in a Wasp app to verify that it works as intended.

- 🧪 Tests and apps:

  - [ ] I added **unit tests** for my change. 
  - [ ] _(if you fixed a bug)_ I added a **regression test** for the bug I fixed. 
  - [ ] _(if you added/updated a feature)_ I added/updated **e2e tests** in `examples/kitchen-sink/e2e-tests`.
  - [ ] _(if you added/updated a feature)_ I updated the **starter templates** in `waspc/data/Cli/templates`, as needed.
  - [ ] _(if you added/updated a feature)_ I updated the **example apps** in `examples/`, as needed.
    - [ ] _(if you updated `examples/tutorials`)_ I updated the tutorial in the docs (and vice versa).

- 📜 Documentation:

  - [ ] _(if you added/updated a feature)_ I **added/updated the documentation** in `web/docs/`.

- 🆕 Changelog: _(if change is more than just code/docs improvement)_
  - [ ] I updated `waspc/ChangeLog.md` with a **user-friendly** description of the change.
  - [ ] _(if you did a breaking change)_ I added a step to the current **migration guide** in `web/docs/migration-guides/`.
  - [ ] I **bumped the `version`** in `waspc/waspc.cabal` to reflect the changes I introduced.

### CODE REVIEW

File: waspc/tools/install_packages_to_data_dir.ps1 [hunk 1]
```diff
@@ -4,8 +4,19 @@
...
+# Clean up old packages in data dir.
+if ([string]::IsNullOrEmpty($dataPackagesDir)) {
+    throw "dataPackagesDir must be set before cleanup"
+}
+if (Test-Path $dataPackagesDir) {
+    Remove-Item -Path $dataPackagesDir -Recurse -Force
+}
+New-Item -Path $dataPackagesDir -ItemType Directory -Force | Out-Null
+
+$packageDirs = Get-ChildItem -Path "$wascpDir/packages" -Directory
```
Comment (infomiho) [inline waspc/tools/install_packages_to_data_dir.ps1:16]: I did the same logical changes in the main PR to the bash script:
- Introduce the `waspcDir` variable
- Check if `dataPackagesDir` variable exists
- Clean up the packages upfront

File: .github/workflows/ci-waspc-test.yaml [hunk 2]
```diff
@@ -87,6 +87,11 @@ jobs:
         run: ./tools/install_packages_to_data_dir.ps1
         shell: pwsh
 
+      - name: Compile Wasp libs and move them into the Cabal data dir on Windows
+        if: runner.os == 'Windows'
+        run: ./tools/install_libs_to_data_dir.ps1
+        shell: pwsh
+
```
Comment (Martinsos) [inline .github/workflows/ci-waspc-test.yaml]: We could compress these into just two steps? I think it less noise then.
I also tried to improve the titles a bit (shorter, close to what scripts are named).

```suggestion
     - name: [Lnx&Mac] Install TS packages and Wasp libs to Cabal data dir
       if: runner.os == 'Linux' || runner.os == 'macOS'
       run: |
         ./tools/install_packages_to_data_dir.sh
         ./tools/install_libs_to_data_dir.sh

     - name: [Windows] Install TS packages and Wasp libs to Cabal data dir
       if: runner.os == 'Windows'
       run: |
         ./tools/install_packages_to_data_dir.ps1
         ./tools/install_libs_to_data_dir.ps1
       shell: pwsh
```

File: waspc/tools/install_packages_to_data_dir.ps1 (hunk 1 above)
Comment (Martinsos) [reply waspc/tools/install_packages_to_data_dir.ps1:16]: Aha ok, so this is just following up on the improvements on the bash script, ok makes sense!

File: waspc/tools/install_packages_to_data_dir.ps1 [hunk 3]
```diff
@@ -4,8 +4,19 @@
...
 # Gets the directory of where this script lives.
 $dir = Split-Path -Parent $MyInvocation.MyCommand.Path
+$wascpDir = Resolve-Path "$dir/.."
+$dataPackagesDir = Join-Path $wascpDir "data/packages"
 
-$packageDirs = Get-ChildItem -Path "$dir/../packages" -Directory
+# Clean up old packages in data dir.
+if ([string]::IsNullOrEmpty($dataPackagesDir)) {
+    throw "dataPackagesDir must be set before cleanup"
+}
```
Comment (Martinsos) [inline waspc/tools/install_packages_to_data_dir.ps1]: Do we need this if we hvae Test-Path thing below? I am not sure what TestPath does, but it sounds like it will fail if string is empty or null.

File: waspc/tools/install_libs_to_data_dir.ps1 [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+# Gets the directory of where this script lives.
+$dir = Split-Path -Parent $MyInvocation.MyCommand.Path
+$wascpDir = Resolve-Path "$dir/.."
+$dataLibsDir = Join-Path $wascpDir "data/Generator/libs"
+
+# Clean up old libs.
+if ([string]::IsNullOrEmpty($dataLibsDir)) {
+    throw "dataLibsDir must be set before cleanup"
+}
```
Comment (Martinsos) [inline waspc/tools/install_libs_to_data_dir.ps1]: Redundant due to Test-Path?

File: waspc/tools/install_libs_to_data_dir.ps1 [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+if (Test-Path $dataLibsDir) {
+    Remove-Item -Path $dataLibsDir -Recurse -Force
+}
+New-Item -Path $dataLibsDir -ItemType Directory -Force | Out-Null
+
+# Build and copy libs to data dir.
```
Comment (Martinsos) [inline waspc/tools/install_libs_to_data_dir.ps1]: This comment is ok if there isno newline after `$libDirs`, otherwise it is confusing, so I would say remove it or that newline.

File: .github/workflows/ci-waspc-test.yaml (hunk 2 above)
Comment (infomiho) [reply .github/workflows/ci-waspc-test.yaml]: Great idea 👍 I'll do that

File: waspc/tools/install_packages_to_data_dir.ps1 (hunk 3 above)
Comment (infomiho) [reply waspc/tools/install_packages_to_data_dir.ps1]: It will fail if the string is empty, so we don't really need to have the string check, you are right 👍 

```
PS C:\Users\Mihovil> Test-Path "/"
True
PS C:\Users\Mihovil> Test-Path "."
True
PS C:\Users\Mihovil> Test-Path ""
Test-Path : Cannot bind argument to parameter 'Path' because it is an empty string.
At line:1 char:11
+ Test-Path ""
+           ~~
    + CategoryInfo          : InvalidData: (:) [Test-Path], ParameterBindingValidationException
    + FullyQualifiedErrorId : ParameterArgumentValidationErrorEmptyStringNotAllowed,Microsoft.PowerShell.Commands.Test
   PathCommand
```
